example_cpp_smart_card_client_app/src/chrome_certificate_provider/bridge-backend.js

```diff
@@ -41,15 +41,66 @@ var NACL_INCOMING_REQUESTER_NAME = 'certificate_provider_nacl_incoming';
 /** @const */
 var NACL_OUTGOING_REQUESTER_NAME = 'certificate_provider_nacl_outgoing';
 
-/** @const */
+/**
+ * @const Function called in the NaCl module when handling
+ * onCertificatesUpdateRequested/onCertificatesRequested events.
+ */
 var HANDLE_CERTIFICATES_REQUEST_FUNCTION_NAME = 'HandleCertificatesRequest';
 
-/** @const */
-var HANDLE_SIGN_DIGEST_REQUEST_FUNCTION_NAME = 'HandleSignDigestRequest';
+/**
+ * @const Function called in the NaCl module when handling
+ * onSignatureRequested/onSignDigestRequested events.
+ */
+var HANDLE_SIGNATURE_REQUEST_FUNCTION_NAME = 'HandleSignatureRequest';
 
 /** @const */
 var GSC = GoogleSmartCard;
 
+/**
+ * Mapping from the |Hash| enum values to the |Algorithm| ones.
+ * @type {!Map<!chrome.certificateProvider.Hash,
+ *             !chrome.certificateProvider.Algorithm>}
+ */
+const HASH_TO_ALGORITHM_MAPPING = new Map([
+  [chrome.certificateProvider.Hash.MD5_SHA1,
```

Comment:
This part is not ideal - this will crash if the App is started in the environment where there's no `chrome.certificateProvider` API (e.g., on desktop or in unit tests). I'm open to ideas - currently planning to do the boring copy-paste with two switches, one in `getAlgorithmFromHash()` and another in `getHashFromAlgorithm()`.